app/scripts/app.js

```diff
@@ -481,6 +481,68 @@ angular
       LabelFilter.readPersistedState();
     });
   })
+  .run(function($location, $rootScope, $uibModal) {
+    if ($location.path() === '/logout' || $location.path() === '/error') {
+      return false;
+    }
+    var isLocalLogoutModalShown = false;
+    var updateLastInteraction = function() {
+      localStorage.setItem('lastInteraction', new Date());
+    };
+    var updateLogOut = function(bool) {
+      localStorage.setItem('logOut', bool);
+    };
+    var updateCloseAllModals = function(bool) {
+      localStorage.setItem('closeAllModals', bool);
+    };
+    var logout = function() {
+      $location.url('/logout');
+    };
+    var showLogoutModal = function() {
+      if (isLocalLogoutModalShown) {
+        return false;
+      }
+      isLocalLogoutModalShown = true;
+      updateCloseAllModals(false);
+      var modalInstance = $uibModal.open({
+        animation: true,
+        templateUrl: 'views/modals/logout.html',
+        controller: 'LogoutModalController',
+      });
+      modalInstance.result.then(function(event) {
+        if (event === 'logout') {
+          updateLogOut(true);
+          logout();
+        } else {
+          isLocalLogoutModalShown = false;
+          updateCloseAllModals(true);
+        }
+      }, function () {
+        updateLastInteraction();
+        isLocalLogoutModalShown = false;
+        updateCloseAllModals(true);
+      });
+    };
+
+    updateLastInteraction();
+    updateLogOut(false);
+    updateCloseAllModals(false);
+
+    $(document).bind("click keydown keypress", _.throttle(updateLastInteraction, 500));
+
+    setInterval(function(){
+      var lastInteraction = Date.parse(localStorage.getItem('lastInteraction'));
+      var newDate = new Date();
+      console.log(newDate - lastInteraction);
+      if (newDate - lastInteraction > 10000) {
+        showLogoutModal();
+      }
+      if (localStorage.getItem('logOut') === 'true') {
+        logout();
```

Comment:
I'm trying to decide if its worth tracking this to force all the other tabs to go to the logout page immediately... the token is no longer valid as soon as any of the tabs go through logout. So any API interaction happening in the background on the other tabs would trigger a redirect to the login flow, and if no API interaction happened then they would just get to the logout page on their own when their own timer fired. 

@spadgett thoughts?

Reply:
Maybe just change the the message in the dialog, after the timeout.. something like:

![login](https://user-images.githubusercontent.com/1668218/33209573-c488d2be-d116-11e7-8dae-b85f99672c59.png)

That would make the logic more simple and we will solve the "redirect and notify" user issue, cause user will see that he was logged out.
@jwforres @ncameronbritt @serenamarie125  thoughts ?